src/pal/src/cruntime/misctls.cpp

```diff
@@ -150,250 +164,155 @@ See MSDN for more information.
+    // Round if needed
+    //
 
-    if ( nTempBufferLength >= count && count >= 0 )
+    if (mantissaIndex >= exponentIndex || lpStartOfReturnBuffer[mantissaIndex] < '5')
     {
-        /* Determine whether I need to round up. */
-        if ( *(lpReturnBuffer + count) >= '5' )
-        {
-            CHAR cNumberToBeRounded;
-            if ( count != 0 )
-            {
-                cNumberToBeRounded = *(lpReturnBuffer + count - 1);
-            }
-            else
-            {
-                cNumberToBeRounded = *lpReturnBuffer;
-            }
-            
-            if ( cNumberToBeRounded < '9' )
-            {
-                if ( count > 0 )
-                {
-                    /* Add one to the character. */
-                    (*(lpReturnBuffer + count - 1))++;
-                }
-                else
-                {
-                    if ( cNumberToBeRounded >= '5' )
-                    {
-                        (*dec)++;
-                    }
-                }
-            }
-            else
-            {
-                LPSTR lpRounding = NULL;
-
-                if ( count > 0 )
-                {
-                    lpRounding = lpReturnBuffer + count - 1;
-                }
-                else
-                {
-                    lpRounding = lpReturnBuffer + count;
-                }
-
-                while ( cNumberToBeRounded == '9' )
-                {
-                    cNumberToBeRounded = *lpRounding;
-                    
-                    if ( cNumberToBeRounded == '9' )
-                    {
-                        *lpRounding = '0';
-                        lpRounding--;
-                    }
-                }
-                
-                if ( lpRounding == lpStartOfReturnBuffer )
-                {
-                    /* Overflow. number is a whole number now. */
-                    *lpRounding = '1';
-                    memset( ++lpRounding, '0', count);
-
-                    /* The decimal has moved. */
-                    (*dec)++;
-                }
-                else
-                {
-                    *lpRounding = ++cNumberToBeRounded;
-                }
-            }
-        }
-        else
-        {
-            /* Get rid of the preceding 0 */
-            lpStartOfReturnBuffer++;
-        }
+        return lpStartOfReturnBuffer; // rounding is not needed
     }
 
-    if ( *lpStartOfReturnBuffer == '0' )
+    i = count - 1;
+    while (lpStartOfReturnBuffer[i] == '9' && i > 0)
     {
-        lpStartOfReturnBuffer++;
+        lpStartOfReturnBuffer[i] = '0';
+        i--;
     }
 
-    if ( count >= 0 )
+    if (i == 0 && lpStartOfReturnBuffer[i] == '9')
     {
-        *(lpStartOfReturnBuffer + count) = '\0';
+        lpStartOfReturnBuffer[i] = '1';
+        (*dec)++;
     }
     else
     {
-        *lpStartOfReturnBuffer = '\0';
-    }
-
-done:
-
-    LOGEXIT( "_ecvt returning %p (%s)\n", lpStartOfReturnBuffer , lpStartOfReturnBuffer );
```

Comment:
As in a comment above, we should either keep these (because they are used in some scenarios) or we should remove the matching `PERF_ENTRY(_ecvt);` calls on L167.